src/einsum/lib.py

```diff
@@ -77,36 +78,26 @@ class EinsumSpec:
     # output is an instance of EinsumOutputSpec
     output: EinsumOutputSpec
 
-EINSUM_LETTERS_UPPER = 26 # oounts upper case letters A-Z
-EINSUM_LETTERS_LOWER = 26 # counts lower case letters a-z
-EINSUM_LETTERS = EINSUM_LETTERS_UPPER + EINSUM_LETTERS_LOWER
+EINSUM_LETTERS_UPPER = string.ascii_uppercase # A-Z
+EINSUM_LETTERS_LOWER = string.ascii_lowercase # a-z
+EINSUM_LETTERS = EINSUM_LETTERS_UPPER + EINSUM_LETTERS_LOWER # A-Za-z
 
 def einsum_index(letter: str) -> int:
-    if "A" <= letter and letter <= "Z":
-        x = ord(letter) - ord("A")
-        assert 0 <= x < EINSUM_LETTERS_UPPER
-        return x
-    elif "a" <= letter and letter <= "z":
-        x = ord(letter) - ord("a")
-        assert 0 <= x < EINSUM_LETTERS_LOWER
-        return EINSUM_LETTERS_UPPER + x
-    assert False, f"index '{letter}' ({ord(letter)}) is not a letter"
+    assert letter in EINSUM_LETTERS, f"index '{letter}' ({ord(letter)}) is not a valid einsum letter"
```

Comment:
the assert and result computation is nice and concise but is it slower than than the old logic?

would it be faster/worthwhile to define a dictionary that maps the 52 letters to their indices?